### PR TITLE
#CU-2e13evn xcmp to karura

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
## Description
* xcmp tooling starter
* end to end xcmp assets from karura end to end
* improve runs of local chains
* docs on assets registry

## Checklist

- [x] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [x] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2e13evn Submit transfer ceremonies proposals](https://app.clickup.com/t/2e13evn)

File: Cargo.toml [hunk 1]
```diff
@@ -33,20 +33,21 @@ exclude = [
...
+  # "runtime/primitives",
+  # "runtime/picasso",
+  # "runtime/dali",
+  # "runtime/composable",
+  # "utils/common",
+  # "utils/price-feed",
+  # "utils/collator-sidecar",
+  # "simnode/simnode-node",
+  # "simnode/simnode-tests",
+  # "simnode/common",
```
Comment (KaiserKarel) [inline Cargo.toml:50]: Can we remove this?

File: frame/assets-registry/src/lib.rs [hunk 2]
```diff
@@ -163,7 +163,13 @@ pub mod pallet {
...
 	impl<T: Config> Pallet<T> {
 		/// creates asset using `CurrencyFactory`,
-		/// raises `AssetRegistered` event
+		/// Raises `AssetRegistered` event
+		///
+		/// # Parameters:
+		/// `ratio` - given unit of foreign asset id, how much of native assets it would be given.
+		/// One to one is 10^18 integer. 10*10^8 will tell that for 1 foreign asset, 10 local native
+		/// will be given. `decimals` - remote number of decimals on other chain
+		/// `ed` - same meaning as in `CurrencyFactory`
```
Comment (KaiserKarel) [inline frame/assets-registry/src/lib.rs:172]: What do we mean by `given`? The fee conversion?

**Review by KaiserKarel — DISMISSED** [review]
Is there a way for us to generate utils/xcmp/src/generated/rococo_relay_chain.rs at build time, so we can avoid reviewing generated code, but instead review the code generation?

File: rfcs/0004-xcmp-picasso-karura.md [hunk 3]
```diff
@@ -4,56 +4,139 @@ This is guide on how to open XCMP channel to Karura and register PICA token here
...
-5. Make PICA transferable to Karura
+- [Overview](#overview)
+  - [Prerequisites](#prerequisites)
+  - [Picasso to Karura](#picasso-to-karura)
+  - [Accept Picasso to Karura](#accept-picasso-to-karura)
+  - [Karura to Picasso](#karura-to-picasso)
+  - [Accept Karura on Picasso](#accept-karura-on-picasso)
+  - [Karura assets on Picasso](#karura-assets-on-picasso)
+  - [Picasso tokens on Karura](#picasso-tokens-on-karura)
+  - [Reference](#reference)
```
[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply rfcs/0004-xcmp-picasso-karura.md:15]: it was kind of right by markdown linter. linter tells there could be one and only one #, all other should be ##. and one is h1 and other h2, so they outlined by markdown table generator tooling. so it is kind of right. so not pleasant may be, but tooling tells such and i am ok with autoformatting like that.

File: scripts/polkadot-launch/README.md [hunk 4]
```diff
@@ -1,85 +1,34 @@
...
-	yarn
-	```
-
-4. run all
-
-	```bash
-	yarn composable
-	```
+```bash
+yarn composable
```
Comment (benluelo) [inline scripts/polkadot-launch/README.md:16]:
```suggestion
yarn && yarn composable
```

File: Cargo.toml [hunk 5]
```diff
@@ -32,22 +32,7 @@ exclude = [
...
-  "runtime/dali",
-  "runtime/composable",
-  "utils/common",
-  "utils/price-feed",
-  "utils/collator-sidecar",
-  "simnode/simnode-node",
-  "simnode/simnode-tests",
-  "simnode/common",
-]
+members = ["utils/xcmp"]
```
[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply Cargo.toml:35]: fixed. sorry

File: scripts/polkadot-launch/README.md [hunk 6]
```diff
@@ -1,85 +1,34 @@
...
-	yarn
-	```
-
-5. run all
-
-	```bash
-	yarn composable_and_basilisk
-	```
+```bash
+yarn composable_and_basilisk
```
Comment (benluelo) [inline scripts/polkadot-launch/README.md:28]:
```suggestion
yarn && yarn composable_and_basilisk
```

**Review by benluelo — APPROVED** [review]

File: scripts/polkadot-launch/README.md (hunk 6 above)
Comment (dzmitry-lahoda) [reply scripts/polkadot-launch/README.md:28]: i will remove basilisk and replace with acala next time. so will do install and run, so not now. thans for suggestion.

**Review by haroldsphinx — APPROVED** [review]

**Pull request merged**